optimum/exporters/executorch/integrations.py

```diff
@@ -24,15 +24,274 @@
     PreTrainedModel,
     StaticCache,
     T5ForConditionalGeneration,
+    VoxtralForConditionalGeneration,
     WhisperForConditionalGeneration,
 )
 from transformers.generation.configuration_utils import GenerationConfig
+from transformers.integrations.executorch import TorchExportableModuleForDecoderOnlyLM, sdpa_mask_without_vmap
+from transformers.masking_utils import AttentionMaskInterface
+from transformers.modeling_utils import AttentionInterface
 
 from optimum.executorch.attentions.custom_sdpa import get_custom_sdpa_for_ring_kv_cache
 
 from .utils import save_config_to_constant_methods
 
 
+class VoxtralEncoderExportableModule(torch.nn.Module):
+    """
+    Subgraph which handles all of the audio-related work: encoder, multimodal projection, combinining with text tokens.
+    The result of this subgraph should stream directly into the decoder subgraph.
+    """
+
+    def __init__(self, model: torch.nn.Module):
+        super().__init__()
+        self.audio_encoder = model.audio_tower
+        self.mm_projector = model.multi_modal_projector
+        self.intermediate_size = model.config.audio_config.intermediate_size
+        self.audio_token_id = model.config.audio_token_id
+        self.metadata = save_config_to_constant_methods(model.config.text_config, model.generation_config)
+
+    def prepare_export_inputs(self):
+        chunk_length = (
+            self.audio_encoder.config.max_source_positions
+            * self.audio_encoder.conv1.stride[0]
+            * self.audio_encoder.conv2.stride[0]
+        )
+        input_features = torch.rand(3, 128, chunk_length)  # (bsz, features, seq_len)
```

Comment:
where do these values come from? I think this is specific per model. Like voxtral has its own preprocessor_config.json. Is this not avaialbe via some config?

Reply:
or this 128 sounds like num_mel_bins in the config